strawberry/annotation.py

```diff
@@ -124,11 +127,8 @@ def set_namespace_from_field(self, field: StrawberryField) -> None:
         self.namespace = module.__dict__
 
     def create_concrete_type(self, evaled_type: type) -> type:
-        if _is_object_type(evaled_type):
-            type_definition: TypeDefinition
-            type_definition = evaled_type._type_definition  # type: ignore
-            return type_definition.resolve_generic(evaled_type)
-
+        if has_strawberry_object(evaled_type):
```

Comment:
I'm not 100% I like this name, `has_strawberry_object` somewhat implies that anything could have a StrawberryObject 😊

I think `is_strawberry_object` might be better 😊

/cc @BryceBeagle

Reply:
It follows attrs `has_attrs` but I'm cool with both.

Reply:
I called it `StrawberryObjectDefinition` because of GraphQL calls "types" objects, [Input Object](http://spec.graphql.org/October2021/#sec-Input-Object-Values) *type* [Object type](http://spec.graphql.org/October2021/#sec-Objects) [Input object](http://spec.graphql.org/October2021/#sec-Input-Objects) and [interfaces](http://spec.graphql.org/October2021/#sec-Interfaces) all of them are closely related because they have fields that can have other types. Also as far as GraphQL concern enum / scalar are also "types"

Reply:
> I personally prefer is_strawberry_object as well.

So how would you call `get_object_definition`? `get_strawberry_object_definition`?